youtube_dl/extractor/vidlii.py

```diff
@@ -0,0 +1,129 @@
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.vidlii.com/watch?v=vBo2IcrwOkO',
+        'md5': 'b42640a596b4dc986702567d49268963',
+        'info_dict': {
+            'id': 'vBo2IcrwOkO',
+            'title': '(OLD VIDEO) i like youtube!!',
+            'description': 'Original upload date:<br />\nMarch 10th 2011<br />\nCredit goes to people who own content in the video',
+            'thumbnail': 'https://www.vidlii.com/usfi/thmp/vBo2IcrwOkO.jpg',
+            'uploader': 'MyEditedVideoSpartan',
+            'url': 'https://cdn.vidlii.com/videos/vBo2IcrwOkO.mp4',
+            'uploader_url': 'https://www.vidlii.com/user/MyEditedVideoSpartan',
+            'upload_date': '20171011',
+            'categories': 'Film & Animation',
+            'tags': None,
+            'duration': 34,
+            'view_count': int,
+            'comment_count': int,
+            'average_rating': float,
+            'type': 'video',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # extract basic properties of video
+        title = self._html_search_regex(r'<h1>(.+?)</h1>', webpage, 'title', default=None) or self._html_search_regex(
+            r'<title>([^<]+?)</title>', webpage, 'title', default="").replace(
+            " - VidLii", "") or self._html_search_meta('twitter:title', webpage, 'title',
+                                                       default="").replace(" - VidLii", "")
+
+        description = strip_or_none(get_element_by_id('des_text', webpage))
+
+        uploader = self._html_search_regex(r'<div[^>]+class="wt_person"[^>]*>(?:[^<]+)<a href="\/user\/[^>]*?>([^<]*?)<',
+                                           webpage, 'uploader', default=None) or self._html_search_regex(
+            r'<img src="[^>]+?class=["\']avt2\s*["\'][^>]+?alt=["\']([^"\']+?)["\']',
+            webpage, 'uploader', default=None)
+
+        url = self._html_search_regex(r'videoInfo[\s]*=[\s]*{[^}]*src:[\s]*(?:"|\')([^"]*?)(?:"|\')', webpage, 'url',
+                                      default=None)
+
+        # get additional properties
+        uploader_url = urljoin("https://www.vidlii.com/user/", uploader)
+
+        # returns date as YYYYMMDD
+        upload_date = self._html_search_meta('datePublished', webpage, 'upload_date', default="").replace("-", "")
+
+        categories = self._html_search_regex(
+            r'<div>Category:\s*<\/div>[\s\r]*<div>[\s\r]*<a href="\/videos\?c=[^>]*>([^<]*?)<\/a>', webpage, 'categories',
+            default=None)
+        tags = re.findall(r'<a href="/results\?q=[^>]*>[\s]*([^<]*)</a>', webpage) or None
+        duration = int_or_none(self._html_search_meta('video:duration', webpage, 'duration', default=False)) or int_or_none(
+            self._html_search_regex(r'videoInfo[^=]*=[^{]*{[^}]*dur:([^,}]*?),', webpage, 'duration', default=None))
```

Comment:
Code duplication.

Reply:
Only whitespace is allowed between `videoInfo` and `=`.

Reply:
`_parse_json`.

Reply:
Cannot use ```_parse_json``` here because the information on the video site is no proper JSON, it is a Javascript dictionary (keys do not have required ```"```). 

FIXED: Code duplication removed, changed RE to only allow whitespace characters

Reply:
`js_to_json`.

Reply:
FIXED: Used ```js_to_json``` and ```_parse_json``` to convert ```videoInfo``` into dictionary.